### PR TITLE
[OSDOCS-6831]: Adding etcd recovery docs for hosted control planes

### PR DESCRIPTION
**Pull request by lahinson**
Version(s): 4.14

Issue: https://issues.redhat.com/browse/OSDOCS-6831

Link to docs preview: https://63983--docspreview.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-backup-restore-dr#hosted-etcd-non-disruptive-recovery

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/hosted-cluster-single-node-recovery.adoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
+// Module included in the following assembly:
+//
+// * hcp-backup-restore-dr.adoc
+
+:_content-type: PROCEDURE
+[id="hosted-cluster-single-node-recovery_{context}"]
+= Recovering an etcd member for a hosted cluster
+
+If one etcd member of a 3-node cluster has corrupted data, it might fail. To recover the etcd member, complete the following steps.
```
Comment (csrwng) [inline modules/hosted-cluster-single-node-recovery.adoc:9]: I think a better wording for the first sentence would be:
`An etcd member of a 3-node cluster might fail due to corrupted or missing data.`

**Review by csrwng — COMMENTED** [review]
Thank you @lahinson 
Just one comment, otherwise lgtm

File: modules/hosted-cluster-single-node-recovery.adoc (hunk 1 above)
Comment (lahinson) [reply modules/hosted-cluster-single-node-recovery.adoc]: Thanks! I incorporated that revision.

File: hosted_control_planes/hcp-backup-restore-dr.adoc [hunk 2]
```diff
@@ -11,10 +11,20 @@ If you need to back up and restore etcd on a hosted cluster or provide disaster
 :FeatureName: Hosted control planes
 include::snippets/technology-preview.adoc[]
 
+[id="hosted-etcd-non-disruptive-recovery"]
+== Recovering etcd pods for hosted clusters
+
+In hosted clusters, etcd pods run as part of a stateful set. The stateful set relies on persistent storage to store etcd data for each member. In a highly available control plane, the size of the stateful set is 3, and each member has its own persistent volume claim.
```
Comment (mburke5678) [inline hosted_control_planes/hcp-backup-restore-dr.adoc:17]: Per ISG, spell out numbers zero through nine.
Do we need to say three what? Pods? Members? (Are these different?)

Comment (lahinson) [reply hosted_control_planes/hcp-backup-restore-dr.adoc:17]: Good catch! I'll update the text to clarify that it's three members.

File: modules/hosted-cluster-single-node-recovery.adoc [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+An etcd member of a 3-node cluster might fail because of corrupted or missing data. To recover the etcd member, complete the following steps.
+
+.Procedure
+
+. If you need to confirm that the etcd member is failing, enter the following command:
++
+[source,terminal]
+----
+$ oc get pods -l app=etcd -n $CONTROL_PLANE_NAMESPACE
+----
```
Comment (mburke5678) [inline modules/hosted-cluster-single-node-recovery.adoc:18]: As I understand, this suggests an environment variable (right?). Should this be a variable? 

`oc get pods -l app=etcd -n  <control_plane_namespace>`

File: modules/hosted-cluster-single-node-recovery.adoc [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
++
+The output looks like this example if the etcd member is failing:
```
Comment (mburke5678) [inline modules/hosted-cluster-single-node-recovery.adoc:20]: Is _looks_ anthropomorphic? 

> The output appears similar to the following if the etcd member is failing:

File: modules/hosted-cluster-single-node-recovery.adoc [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+.Example output
+[source,terminal]
+----
+NAME     READY   STATUS             RESTARTS     AGE
+etcd-0   2/2     Running            0            64m
+etcd-1   2/2     Running            0            45m
+etcd-2   1/2     CrashLoopBackOff   1 (5s ago)   64m
+----
+
+. Delete the persistent volume claim of the etcd member and the pod by entering the following command:
```
Comment (mburke5678) [inline modules/hosted-cluster-single-node-recovery.adoc:31]: Is this too obvious to be useful?
```suggestion
. Delete the persistent volume claim of the failing etcd member and the pod by entering the following command:
```

File: modules/hosted-cluster-etcd-status.adoc [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+. Set up the etcdctl environment by entering the following commands:
++
+[source,terminal]
+----
+export ETCDCTL_API=3
+export ETCDCTL_CACERT=/etc/etcd/tls/etcd-ca/ca.crt
+export ETCDCTL_CERT=/etc/etcd/tls/client/etcd-client.crt
+export ETCDCTL_KEY=/etc/etcd/tls/client/etcd-client.key
+export ETCDCTL_ENDPOINTS=https://etcd-client:2379
+----
```
Comment (mburke5678) [inline modules/hosted-cluster-etcd-status.adoc:29]: Do we need to add the prompt for these commands (`sh-4.4#`)? 
Should these commands be in separate code blocks, [per the guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output)?

File: modules/hosted-cluster-etcd-status.adoc [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+. Print the endpoint status for each cluster member by entering the following command:
++
+[source,terminal]
+----
+etcdctl endpoint health --cluster -w table
+----
```
Comment (mburke5678) [inline modules/hosted-cluster-etcd-status.adoc:36]: Prompt?

**Review by mburke5678 — CHANGES_REQUESTED** [review]
A few nits. Otherwise LGTM

File: modules/hosted-cluster-single-node-recovery.adoc (hunk 3 above)
Comment (lahinson) [reply modules/hosted-cluster-single-node-recovery.adoc:18]: Yep! I'll update that.

File: modules/hosted-cluster-single-node-recovery.adoc (hunk 4 above)
Comment (lahinson) [reply modules/hosted-cluster-single-node-recovery.adoc:20]: I don't think _looks_ is anthropomorphic unless the intention is to say that the output is looking at something. I can change it to "The output resembles this example..."

File: modules/hosted-cluster-etcd-status.adoc (hunk 6 above)
Comment (lahinson) [reply modules/hosted-cluster-etcd-status.adoc:44]: Yep. I just updated those commands and the one in step 3.

Comment (mburke5678) [reply modules/hosted-cluster-etcd-status.adoc:44]: I believe the commands here should be `sh-4.4# export ..` The pod itself doesn't have the `oc` client installed.
This is what I am getting:
```
sh-4.4# oc export ETCDCTL_CACERT=/etc/etcd/tls/etcd-ca/ca.crt
sh: oc: command not found
sh-4.4# export ETCDCTL_CACERT=/etc/etcd/tls/etcd-ca/ca.crt
sh-4.4# 
```

File: modules/hosted-cluster-etcd-status.adoc (hunk 7 above)
Comment (mburke5678) [reply modules/hosted-cluster-etcd-status.adoc:51]: Same here:
```
sh-4.4# oc etcdctl endpoint health --cluster -w table
sh: oc: command not found
```

File: modules/hosted-cluster-etcd-status.adoc [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+$ oc export ETCDCTL_CACERT=/etc/etcd/tls/etcd-ca/ca.crt
+----
++
+[source,terminal]
+----
+$oc export ETCDCTL_CERT=/etc/etcd/tls/client/etcd-client.crt
+----
++
+----
```
Comment (mburke5678) [inline modules/hosted-cluster-etcd-status.adoc:37]: [source,terminal]

**Pull request merged**